app/controllers/queue_items_controller.rb

```diff
@@ -12,6 +12,17 @@ def create
     redirect_to my_queue_path
   end
 
+  def update
+    update_queue_item_ratings
+    begin
+      update_queue_positions
```

Comment:
Updating ratings and positions separately is a little bit expensive, you should try to find queue item once and then do the updates, rather than finding it again afterwards and doing another set of updates. Also updating ratings outside of the begin/rescue and positions inside throws in a question - why?

Reply:
I implemented the functions separately because I want the user to keep their ratings even when the corresponding position inputs are not valid.